test/axios.headers.test.js

```diff
@@ -0,0 +1,1993 @@
+/* Copyright (c) 2020, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+let axios = require("axios");
+
+let puppeteer = require("puppeteer");
+let jsdom = require("mocha-jsdom");
+let AuthHttpRequest = require("../index.js").default;
+let AuthHttpRequestFetch = require("../lib/build/fetch").default;
+let AuthHttpRequestAxios = require("../lib/build/axios").default;
+let { interceptorFunctionRequestFulfilled, responseInterceptor } = require("../lib/build/axios");
+let assert = require("assert");
+let {
+    delay,
+    checkIfIdRefreshIsCleared,
+    getNumberOfTimesRefreshCalled,
+    startST,
+    getNumberOfTimesGetSessionCalled,
+    BASE_URL,
+    BASE_URL_FOR_ST,
+    getNumberOfTimesRefreshAttempted,
+    resetAuthHttpRequestFetch
+} = require("./utils");
+const { spawn } = require("child_process");
+let { ProcessState, PROCESS_STATE } = require("../lib/build/processState");
+
+AuthHttpRequest.addAxiosInterceptors(axios);
+/* TODO: 
+    - session should not exist when user's session fully expires - use doesSessionExist & check localstorage is empty
+    - while logged in, test that APIs that there is proper change in id refresh cookie
+    - tests APIs that don't require authentication work after logout - with-credentials don't get sent.
+    - if not logged in, test that API that requires auth throws session expired
+    - Test everything without and without interception
+    - If user provides withCredentials as false or whatever, then app should not add it
+    - Cross origin API requests to API that requires Auth
+    - Cross origin API request to APi that doesn't require auth
+    - Proper change in anti-csrf token once access token resets
+    - Refresh API custom headers are working
+    - allow-credentials should not be sent by our SDK by default.
+*/
+describe.skip("Axios AuthHttpRequest class tests header", function() {
+    jsdom({
+        url: "http://localhost.org"
+    });
+
+    before(async function() {
+        spawn(
+            "./test/startServer",
+            [process.env.INSTALL_PATH, process.env.NODE_PORT === undefined ? 8080 : process.env.NODE_PORT, "header"],
```

Comment:
Why do we need to explicitly set `"header"` when starting the backend server? Shouldn't the `st-auth-mode` header in the request decide that?